src/js/components/Form/__tests__/Form-test.js

```diff
@@ -481,6 +481,85 @@ describe('Form', () => {
       }),
     );
   });
+  test('validate on blur', () => {
+    const Test = () => (
+      <Form
+        validate="blur"
+        onReset={event => console.log(event)}
+        onSubmit={({ value }) => console.log('Submit', value)}
+      >
+        <FormField
+          label="Name"
+          name="name"
+          placeholder="name"
+          required
+          validate={[
+            { regexp: /^[a-z]/i },
+            name => {
+              if (name && name.length === 1) return 'must be >1 character';
+              return undefined;
+            },
+            name => {
+              if (name === 'good')
+                return {
+                  message: 'good',
+                  status: 'info',
+                };
+              return undefined;
+            },
+          ]}
+        />
+
+        <FormField label="Email" name="email" required>
+          <TextInput name="email" type="email" placeholder="email" />
+        </FormField>
+        <Button label="submit" type="submit" />
+      </Form>
+    );
+    const {
+      getByText,
+      getByPlaceholderText,
+      queryAllByText,
+      queryByText,
+    } = render(
+      <Grommet>
+        <Test />
+      </Grommet>,
+    );
+
+    // both fields have required error message
+    fireEvent.click(getByText('submit'));
+    expect(queryAllByText('required')).toHaveLength(2);
+
+    // one fields has required error message
+    fireEvent.change(getByPlaceholderText('name'), {
+      target: { value: 'Input has changed' },
+    });
+    fireEvent.click(getByText('submit'));
+    expect(queryAllByText('required')).toHaveLength(1);
+
+    // fields reset to required error message
+    fireEvent.change(getByPlaceholderText('name'), {
+      target: { value: '' },
+    });
+    fireEvent.click(getByText('submit'));
+    expect(queryAllByText('required')).toHaveLength(2);
```

Comment:
Check if this is needed since it is referring to reset.

Reply:
changes made